src/Trip/Manipulator.ts

```diff
@@ -76,6 +76,42 @@ export function addDaysToTrip(
 	return resolveStickiness(trip, userSettings);
 }
 
+export function addDayToTrip(
+	trip: Trip,
+	dayIndex: number,
+	userSettings: UserSettings | null
+): Trip {
+	checkDayExists(trip, dayIndex);
+	if (!trip.endsOn) {
+		throw new Error('endsOn property in Trip cannot be null');
+	}
+
+	let resultTrip = cloneDeep(trip);
+	resultTrip.days.splice(dayIndex, 0, {
+		itinerary: [],
+		note: null,
+		date: null
+	} as Day);
+	resultTrip.endsOn = resultTrip.endsOn ? addDaysToDate(resultTrip.endsOn, 1) : resultTrip.endsOn;
+	resultTrip.days = decorateDaysWithDate(resultTrip.startsOn, resultTrip.days);
+
+	if (dayIndex > 0 && resultTrip.days[dayIndex - 1].itinerary.length > 0) {
+		const previousDayLastItem = resultTrip.days[dayIndex - 1].itinerary[resultTrip.days[dayIndex - 1].itinerary.length - 1];
```

Comment:
I would split this line to multiple expressions. Eg. `const previousDayLastPlaceIndex`